networkx/algorithms/bipartite/envy_free_matching.py

```diff
@@ -0,0 +1,375 @@
+import itertools as it
```

Comment:
"""
An *envy-free matching* (EFM) in an unweighted bipartite graph G = (X+Y, E) is matching in which no unmatched vertex in X is adjacent to a matched vertex in Y. 
The concept is motivated by problems of fair allocation. Suppose the vertices of X represent people, the vertices of Y represent houses, and an edge between a person x and a house y represents the fact that x is willing to live in y. Then, an EFM is a partial allocation of houses to people such that each house-less person does not envy any person with a house, since he/she does not like any allocated house anyway.

An empty matching is trivially envy-free. A perfect matching, in which all vertices of X are matched, is also envy-free. But there are graphs that do not have a perfect matching, but do have a non-empty envy-free matching.

This module provides the following functions, related to EFM:

* `maximum_envy_free_matching` - receives an unweighted bipartite graph, and computes an EFM of maximum cardinality.
* `minimum_weight_envy_free_matching`  - receives a weighted bipartite graph, and computes an EFM of minimum weight, among all EFM of maximum cardinality.
* `envy_free_matching_partition` - receives a bipartite graph G = (X+Y, E), and computes a partition of its vertices into two subsets: XL+YL and XM+YM, that satisfies some unique features related to EFM (details below)
"""

Reply:
It'd be good to use `node` instead of `vertex` to match networkx. 
Thanks for this!

Reply:
> It'd be good to use `node` instead of `vertex` to match networkx. Thanks for this!

Done